udc_stm32 build failure on stm32f1 and stm32f3 (and possibly others)

**Describe the bug**

drivers/usb/udc/udc_stm32.c fails to build on f1 families since https://github.com/zephyrproject-rtos/zephyr/pull/75129 got merged:

```
zephyrproject/zephyr/drivers/usb/udc/udc_stm32.c:792:37: error: 'USBD_HS_SPEED' undeclared (first use in this function); did you mean 'USBD_FS_SPEED'?
  792 |         if (priv->pcd.Init.speed == USBD_HS_SPEED) {
      |                                     ^~~~~~~~~~~~~
      |                                     USBD_FS_SPEED
```

https://github.com/zephyrproject-rtos/zephyr/pull/75306 out with a proposed fix

**To Reproduce**
`west build -p -b stm32_min_dev@blue samples/subsys/usb/hid-keyboard`

**Expected behavior**
No build errors.

**Impact**
Somewhat of an annoyance if you want to actually use the driver, not a problem at all if you don't.

**Environment (please complete the following information):**
Linux, Zephyr SDK, the commit above

--- a/drivers/usb/udc/udc_stm32.c
+++ b/drivers/usb/udc/udc_stm32.c
@@ -789,9 +789,11 @@ static enum udc_bus_speed udc_stm32_device_speed(const struct device *dev)
 {
 	struct udc_stm32_data *priv = udc_get_private(dev);
 
+#ifdef USBD_HS_SPEED
 	if (priv->pcd.Init.speed == USBD_HS_SPEED) {
 		return UDC_BUS_SPEED_HS;
 	}
+#endif
 
 	if (priv->pcd.Init.speed == USBD_FS_SPEED) {
 		return UDC_BUS_SPEED_FS;

